tests/unit/dogstatsd/test_statsd.py

```diff
@@ -147,6 +147,10 @@ def test_initialization(self):
         initialize(**options)
         self.assertEqual(statsd.host, "myhost")
         self.assertEqual(statsd.port, 1234)
+        self.assertIsNone(statsd.socket)
+        with mock.patch('socket.socket'):
+            statsd.get_socket()
+            statsd.socket.connect.assert_called_with(("myhost", 1234))
```

Comment:
If we keep your current impl of `initialize`, I think these tests don't adequately cover the `target_updated = False` logic flow.

Reply:
True, I will add some tests for those cases